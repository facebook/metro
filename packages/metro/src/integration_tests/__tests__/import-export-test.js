/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @emails oncall+javascript_foundation
 * @format
 */

'use strict';

const Metro = require('../../..');

const execBundle = require('../execBundle');

jest.unmock('cosmiconfig');

jest.setTimeout(30 * 1000);

it('builds a simple bundle', async () => {
  const config = await Metro.loadConfig({
    config: require.resolve('../metro.config.js'),
  });

  const result = await Metro.runBuild(config, {
    entry: 'import-export/index.js',
  });

  const object = execBundle(result.code);
  const cjs = await object.asyncImportCJS;

  expect(object).toMatchSnapshot();
  expect(cjs).toEqual(expect.objectContaining(cjs.default));

  await expect(object.asyncImportCJS).resolves.toMatchSnapshot();
  await expect(object.asyncImportESM).resolves.toMatchSnapshot();
});
