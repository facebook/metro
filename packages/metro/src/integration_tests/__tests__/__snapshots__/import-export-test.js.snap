// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`builds a simple bundle 1`] = `
Object {
  "asyncImportCJS": Promise {},
  "asyncImportESM": Promise {},
  "default": "export-4: FOO",
  "extraData": Object {
    "foo": "export-null: FOO",
    "importStar": Object {
      "default": "export-2: DEFAULT",
      "foo": "export-2: FOO",
      "myFunction": [Function],
    },
    "myDefault": "export-1: DEFAULT",
    "myFoo": "export-1: FOO",
    "myFunction": "export-1: MY_FUNCTION",
    "primitiveDefault": "export-primitive-default: DEFAULT",
    "primitiveFoo": "export-primitive-default: FOO",
  },
  "namedDefaultExported": "export-3: DEFAULT",
}
`;

exports[`builds a simple bundle 2`] = `
Object {
  "default": Object {
    "foo": "export-5: FOO",
  },
  "foo": "export-5: FOO",
}
`;

exports[`builds a simple bundle 3`] = `
Object {
  "default": "export-6: DEFAULT",
  "foo": "export-6: FOO",
}
`;
